sql/core/src/test/scala/org/apache/spark/sql/execution/joins/InnerJoinSuite.scala

```diff
@@ -124,7 +125,8 @@ class InnerJoinSuite extends SparkPlanTest with SharedSQLContext {
         rightPlan: SparkPlan) = {
       val sortMergeJoin = joins.SortMergeJoinExec(leftKeys, rightKeys, Inner, boundCondition,
         leftPlan, rightPlan)
-      EnsureRequirements(spark.sessionState.conf).apply(sortMergeJoin)
+      EnsureRequirements(spark.sessionState.conf)
+          .apply(ProjectExec(sortMergeJoin.output, sortMergeJoin))
```

Comment:
Why we need to change this?

Reply:
In 2.1.1, an extra project causes `WholeStageCodegenExec` to not detect that the expression contains `CodegenFallback`. This is no longer the case. Like I said, there is no longer a good way to test what happens when `CodegenFallback` generates code. If there were, I'd use that here to test the case.

I guess I could add a testing case to `WholeStageCodegenExec` to make sure the code is generated correctly.